NumberFormatException: For input string: "FOLLOW_SYSTEM"

### Problem
I can not start the app

### Environment
K-9 Mail version: master (243dd318cfb64c9b9c16a3903402704ccc4d1f50)

```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.fsck.k9.debug, PID: 10367
    java.lang.RuntimeException: Unable to create application com.fsck.k9.App: org.koin.error.BeanInstanceCreationException: Can't create bean Bean[class=com.fsck.k9.Preferences] due to error :
    	java.lang.NumberFormatException: For input string: "FOLLOW_SYSTEM"
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5925)
        at android.app.ActivityThread.access$1100(ActivityThread.java:200)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1656)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6718)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: org.koin.error.BeanInstanceCreationException: Can't create bean Bean[class=com.fsck.k9.Preferences] due to error :
    	java.lang.NumberFormatException: For input string: "FOLLOW_SYSTEM"
        at org.koin.core.instance.InstanceFactory.createInstance(InstanceFactory.kt:63)
        at org.koin.core.instance.InstanceFactory.retrieveInstance(InstanceFactory.kt:26)
        at org.koin.KoinContext$resolveInstance$$inlined$synchronized$lambda$1.invoke(KoinContext.kt:85)
        at org.koin.KoinContext$resolveInstance$$inlined$synchronized$lambda$1.invoke(KoinContext.kt:23)
        at org.koin.ResolutionStack.resolve(ResolutionStack.kt:23)
        at org.koin.KoinContext.resolveInstance(KoinContext.kt:80)
        at com.fsck.k9.K9$$special$$inlined$inject$2.invoke(KoinComponent.kt:97)
        at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
        at com.fsck.k9.K9.getPreferences(Unknown Source:7)
        at com.fsck.k9.K9.init(K9.kt:346)
        at com.fsck.k9.App.onCreate(App.kt:23)
        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1154)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5920)
        	... 8 more
```

--- a/app/storage/src/main/java/com/fsck/k9/preferences/K9StoragePersister.java
+++ b/app/storage/src/main/java/com/fsck/k9/preferences/K9StoragePersister.java
@@ -35,6 +35,11 @@ public class K9StoragePersister implements StoragePersister {
 
         db.beginTransaction();
         try {
+            if (db.getVersion() > DB_VERSION) {
+                throw new AssertionError("Database downgrades are not supported. " +
+                        "Please fix the database '" + DB_NAME + "' manually or clear app data.");
+            }
+
             if (db.getVersion() < 1) {
                 createStorageDatabase(db);
             } else {

--- a/app/storage/src/main/java/com/fsck/k9/storage/StoreSchemaDefinition.java
+++ b/app/storage/src/main/java/com/fsck/k9/storage/StoreSchemaDefinition.java
@@ -46,6 +46,13 @@ class StoreSchemaDefinition implements SchemaDefinition {
 
         db.beginTransaction();
         try {
+            if (db.getVersion() > DB_VERSION) {
+                String accountUuid = migrationsHelper.getAccount().getUuid();
+                throw new AssertionError("Database downgrades are not supported. " +
+                        "Please fix the account database '" + accountUuid + "' manually or " +
+                        "clear app data.");
+            }
+
             // schema version 29 was when we moved to incremental updates
             // in the case of a new db or a < v29 db, we blow away and start from scratch
             if (db.getVersion() < 29) {

